### DISCUSSION TITLE
How one can set TraceSwitch to enable tracing of WinForms internals

### DISCUSSION THREAD

**Discussion by kant2002** [dotnet/runtime#68753] [Q&A]
For now I create debug build and manually set `Level` of desired `TraceSwitch` to `Verbose` but is there way to control these settings via environment or configuration file?

Separate question. ASP.NET Core provide transparancy of internal workings by instrumenting whole application so logs can provide view of the wokring. Do WinForms can do the same? Maybe provide some integration with `M.E.Configuration`?

**Comment by RussKie** [comment]
According to the [docs](https://docs.microsoft.com/dotnet/api/system.diagnostics.traceswitch?view=net-6.0#remarks) `TraceSwitches` can only be configured via app.config in .NET Framework.
![image](https://user-images.githubusercontent.com/4403806/166172688-a1e3699c-00e0-4842-bbd1-ef3c862ec251.png)

Hopefully @dotnet/area-system-configuration can guide us on how to configure those in .NET.

**Reply by danmoseley** [reply]
@JeremyKuhne likely has more familiarity, as he ported it and it's barely been touched since.

**Reply by JeremyKuhne** [reply]
@danmoseley This is more of a `System.Diagnostics` question. It no longer loads the data from the config file and I don't know that the scenario has any other viable alternative without reimplementing the feature. It might be a little complicated given that `System.Configuration` is not part of the core package. I don't know if the runtime configuration would suffice or not- I've never had a chance to look into it.

**Reply by danmoseley** [reply]
Ah yes. And it seems ther'es already an issue -- https://github.com/dotnet/runtime/issues/23937.

**Reply by kant2002** [reply]
Once again, after this request https://github.com/dotnet/winforms/pull/7180/files#r874218389 to add `Debug.WriteLine`, I would like to either use `TraceSource` or somehow start using other logging capabilities, so logging intents can be separated.
